refactor: add folders and tidy up naming/structure

rjerrems opened the pull request:
This PR adds a business unit folder inside the environment folder for application projects and tidies up the naming.

Right now, projects get created at the top of the hierarchy and there are multiple projects with the same name:

```
org > bu2-p-sample    
org > bu2-p-sample             
org > bu2-p-sample-single
```

This PR changes it to:

```
org > prod > bu2-p-sample-floating
org > prod > bu2-p-sample-private
org > prod > bu2-p-sample-restricted
```

- [x] Refactor business unit 1

cc: @mikelaramie

daniel-cit commented on 4-projects/business_unit_2/dev/example_restricted_shared_vpc_project.tf:
```diff
@@ -30,26 +30,10 @@ module "restricted_shared_vpc_project" {
   # Metadata
-  project_prefix    = "bu2-d-sample"
-  application_name  = "bu2-sample-application2"
+  project_prefix    = "bu2-p-sample-restricted"
```
This one should be `bu2-d-sample-restricted` instead of `bu2-p-sample-restricted`

rjerrems replied:
good catch, I will fix in next push

daniel-cit commented on 4-projects/business_unit_2/dev/example_floating_project.tf:
```diff
@@ -14,21 +14,19 @@
   # Metadata
-  project_prefix    = "bu2-d-sample-single"
-  application_name  = "bu2-sample-application2"
+  project_prefix    = "bu2-p-sample-floating"
```
this one should be  `bu2-d-sample-floating` instead of `bu2-p-sample-floating`

daniel-cit commented on 4-projects/business_unit_2/nonprod/example_floating_project.tf:
```diff
@@ -16,21 +16,19 @@
   # Metadata
-  project_prefix    = "bu2-n-sample-single"
-  application_name  = "bu2-sample-application2"
+  project_prefix    = "bu2-p-sample-floating"
```
this one should be `bu2-n-sample-floating` instead of  `bu2-p-sample-floating`

mikelaramie replied:
The env code should be a variable to avoid this, no?

daniel-cit replied:
This is in the context of the `nonprod` example so the only allowed value is `p`. Removing variables that had a value that cloud not be changed was a fix in previous PRs.

daniel-cit commented on 4-projects/business_unit_2/nonprod/example_restricted_shared_vpc_project.tf:
```diff
@@ -30,26 +30,10 @@ module "restricted_shared_vpc_project" {
   # Metadata
-  project_prefix    = "bu2-n-sample"
-  application_name  = "bu2-sample-application2"
+  project_prefix    = "bu2-p-sample-restricted"
```
this one should be  `bu2-n-sample-restricted` instead of  `bu2-p-sample-restricted`

mikelaramie commented:
Hey all, while I agree with this change and have championed for it before (#teamFolders), it goes against the design in the upcoming whitepaper.  Should we pull up internally around this change?

bharathkkb commented:
@rjerrems we discussed this offline with @mikelaramie and Sean and currently guidance from the paper is not to have another folder level. Optionally enabling this via flag was suggested. 
By default it would be `org > prod > bu2-p-sample-private` but if flag is `true` we do `org > prod > business-unit-2 > bu2-p-sample-private` WDYT?

rjerrems commented:
Hey @mikelaramie @daniel-cit @bharathkkb - I have addressed the feedback in most recent commit, have a look when you get a chance and let me know what you think
